opaque_keys/edx/locator.py

```diff
@@ -86,18 +88,28 @@ class BlockLocatorBase(Locator):
     BLOCK_ALLOWED_ID_CHARS = r'[\w\-~.:%]'
 
     ALLOWED_ID_RE = re.compile(r'^' + Locator.ALLOWED_ID_CHARS + '+$', re.UNICODE)
+    ALLOWED_ID_STRATEGY = strategies.text(
+        alphabet=strategies.characters(
+            whitelist_categories=['Lu', 'Ll', 'Lt', 'Lm', 'Lo', 'Nd', 'No']
+        ) | strategies.sampled_from("-~.:"),
+        min_size=1,
+    )
+
     DEPRECATED_ALLOWED_ID_RE = re.compile(r'^' + Locator.DEPRECATED_ALLOWED_ID_CHARS + '+$', re.UNICODE)
+    DEPRECATED_ALLOWED_ID_STRATEGY = strategies.text(
+        alphabet=strategies.characters(
+            whitelist_categories=['Lu', 'Ll', 'Lt', 'Lm', 'Lo', 'Nd', 'No']
+        ) | strategies.sampled_from("-~.:%"),
+        min_size=1,
+    )
```

Comment:
Yup. I'll move them to the bottom of the class, I think, and more clearly demarcate them.